File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        
        //2,5,6,8,12,15

        int min = Integer.MAX_VALUE, sum = 0, left = 0;

        for (int i=0; i < nums.length; i++ ) {
            
            sum += nums[i];

            while ( sum >= target ) {

                min = Math.min(min, i - left +1);
                sum -= nums[left++];

            }
        }

        return min == Integer.MAX_VALUE ? 0: min;

    }
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.go
func minSubArrayLen(target int, nums []int) int {
    
    left, sum, min := 0, 0, math.MaxInt64

    for i:= 0; i < len(nums); i++ {
        sum += nums[i]

        for target <= sum {
            minLength := i - left +1
            if min > minLength {
                min = minLength
            }
            sum -= nums[left]
            left++
        }
    }
    result := 0

    if min < math.MaxInt64 {
        result = min
    }
    return  result;
}